src/Support/QueueMonitor/QueueMonitorManager.php

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+
+    protected static function jobQueued(JobQueued $event): void
+    {
+        app(QueueMonitor::class)->query()->create([
+            'job_batch_id' => $event->job->batchId ?? null,
+            'job_id' => $event->id,
+            'name' => get_class($event->job),
+            'queue' => $event->job->queue ?: 'default',
+            'state' => Queued::class,
+            'queued_at' => now(),
+            'data' => $data ?? null,
+        ]);
+    }
+
+    protected static function jobProcessing(JobProcessing $event): void
+    {
+        $now = Carbon::now();
+
+        $monitor = app(QueueMonitor::class)
+            ->query()
+            ->where('job_id', $jobId = static::getJobId($event->job))
+            ->where('queue', $event->job->getQueue() ?? config('queue.default'))
+            ->whereState('state', Queued::class)
+            ->firstOrNew();
+
+        $monitor->fill([
+            'job_uuid' => $event->job->uuid(),
+            'job_id' => static::getJobId($event->job),
+            'queue' => $event->job->getQueue() ?: config('queue.default'),
+            'name' => $event->job->resolveName(),
+            'started_at' => $now,
+            'started_at_exact' => $now->format('Y-m-d H:i:s.u'),
+            'attempt' => $event->job->attempts(),
+            'state' => Running::class,
+        ]);
+        $monitor->save();
+
+        // Mark jobs with same job id (different execution) as stale
+        app(QueueMonitor::class)->query()
+            ->whereNot('id', $monitor->id)
+            ->where('job_id', $jobId)
+            ->whereNotState('state', Failed::class)
+            ->whereNull('finished_at')
+            ->each(function (QueueMonitor $monitor) {
+                $monitor->update([
+                    'finished_at' => $now = Carbon::now(),
+                    'finished_at_exact' => $now->format('Y-m-d H:i:s.u'),
+                    'state' => Stale::class,
+                ]);
+            });
+    }
+
+    protected static function jobProcessed(JobProcessed $event): void
+    {
+        static::jobFinished($event->job, Succeeded::class);
+    }
+
+    protected static function jobExceptionOccurred(JobExceptionOccurred $event): void
+    {
+        static::jobFinished($event->job, Failed::class, $event->exception);
+    }
+
+    protected static function jobFailed(JobFailed $event): void
+    {
+        static::jobFinished($event->job, Failed::class);
+    }
+
+    protected static function jobFinished(Job $job, string $state, ?\Throwable $exception = null): void
+    {
+        $monitor = app(QueueMonitor::class)
+            ->query()
+            ->where('job_id', static::getJobId($job))
+            ->where('attempt', $job->attempts())
+            ->orderByDesc('started_at')
+            ->first();
+
+        if (! $monitor) {
+            return;
+        }
+
+        /** @var ShouldBeMonitored $resolvedJob */
+        $resolvedJob = $job->resolveName();
```

Comment:
why directly call the resolveName method, although this method may not exist?
Otherwise why is there a method_exists() check in the shouldBeMonitored() method?